jans-auth-server/common/src/test/java/io/jans/as/common/RedirectUriTest.java

```diff
@@ -0,0 +1,342 @@
+        redirectUri.addResponseParameter(CLIENT_ID, "123");
+        assertTrue(redirectUri.toString().length() > 0, "is Empty");
+    }
+
+    @Test()
+    public void toString_NoResponseParams_sameBaseUri() {
+        String baseUri = "http://redirecturl.com/";
+        RedirectUri redirectUri = new RedirectUri(baseUri);
+        assertEquals(redirectUri.toString(), baseUri);
+    }
+
+    private RedirectUri getRedirectUriTemplateToString() {
+        String clientId = "123";
+        String responseType = "token id_token";
+        String scope = "openid";
+        String state = UUID.randomUUID().toString();
+        String nonce = UUID.randomUUID().toString();
+
+        RedirectUri redirectUri = new RedirectUri("http://redirecturl.com/");
+        redirectUri.addResponseParameter(CLIENT_ID, clientId);
+        redirectUri.addResponseParameter(REDIRECT_URI, redirectUri.getBaseRedirectUri());
+        redirectUri.addResponseParameter(RESPONSE_TYPE, responseType);
+        redirectUri.addResponseParameter(SCOPE, scope);
+        redirectUri.addResponseParameter(STATE, state);
+        redirectUri.addResponseParameter(NONCE, nonce);
+        redirectUri.setAuthorizationCodeLifetime(1000);
+        redirectUri.setBlockEncryptionAlgorithm(null);
+        redirectUri.setNestedKeyId("nki");
+        redirectUri.setResponseMode(null);
+        return redirectUri;
+    }
+
+    private RedirectUri getRedirectUriTemplateGetQueryString(ResponseMode responseMode, KeyEncryptionAlgorithm keyEncryptionAlgorithm, BlockEncryptionAlgorithm blockEncryptionAlgorithm, SignatureAlgorithm signatureAlgorithm) throws CryptoProviderException {
+        RedirectUri redirectUri = new RedirectUri("http://redirecturl.com/", new ArrayList<>(), responseMode);
+        redirectUri.setKeyEncryptionAlgorithm(keyEncryptionAlgorithm);
+        redirectUri.setBlockEncryptionAlgorithm(blockEncryptionAlgorithm);
+        redirectUri.setSignatureAlgorithm(signatureAlgorithm);
+        redirectUri.setIssuer("user123");
+        redirectUri.setKeyId("key123");
+        redirectUri.setSharedSecret("shared_secret");
+        redirectUri.setJsonWebKeys(null);
+        redirectUri.setCryptoProvider(cryptoProvider);
+        return redirectUri;
+    }
+
+    private RSAPublicKey getRSAPublicKey() {
+        return new RSAPublicKey() {
+            @Override
+            public BigInteger getPublicExponent() {
+                return null;
+            }
+
+            @Override
+            public String getAlgorithm() {
+                return null;
+            }
+
+            @Override
+            public String getFormat() {
+                return null;
+            }
+
+            @Override
+            public byte[] getEncoded() {
+                return new byte[0];
+            }
+
+            @Override
+            public BigInteger getModulus() {
+                return null;
+            }
+        };
+    }
```

Comment:
I'm guessing that you can replace this by using `mock` of RSAPublicKey instead.

Reply:
I have replaced it, with a method better resumed.